src/components/auth/AuthActionButton.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { useCallback, useEffect, useState } from 'react';
+
+import { AccountCircle, MoreVert } from '@material-ui/icons';
+import { Button, useMediaQuery } from '@material-ui/core';
+import { useHistory } from 'react-router-dom';
+import { useSnackbar } from 'notistack';
+
+import { LoadingOverlay } from '@app/contexts/ui/LoadingOverlay';
+import { useModal } from '@app/contexts/ui/Modal';
+import { useUser } from '@app/contexts/services/User';
+
+import AuthModal from '../auth/AuthModal';
+import IconButtonMenu from '../common/IconButtonMenu';
+
+import useStyles from './_jss/AuthActionButton.jss';
+
+const AuthActionButton: React.FC = () => {
+  const [isLoading, setLoading] = useState(false);
+  const classes = useStyles();
+  const isMobile = useMediaQuery('(max-width:400px)');
+  const modalContext = useModal();
+  const { user, signOut } = useUser();
+  const snackbarContext = useSnackbar();
+  const history = useHistory();
+
+  const discloseAuthModal = useCallback(
+    (option: 'sign-in' | 'sign-up', dismissCallback?: () => void): void => {
+      modalContext?.disclose(<AuthModal initialVariant={option} />, {
+        disablePaper: true,
+        disableTransition: true,
+        onDismiss: dismissCallback,
+      });
+    },
+    [modalContext],
```

Comment:
to avoid impacting dependency arrays using `discloseAuthModal` we should switch to 
```suggestion
    [modalContext.disclose],
``` 

or better yet just destructure `disclose` off the context value. Otherwise `discloseAuthModal` will be a different function between opening/closing the modal.

Reply:
ec5c2bfd56fe50c13a81d66034ea1ba4f11e659e